libs/ui-components/src/utils/patch.ts

```diff
@@ -30,3 +35,104 @@ export const appendJSONPatch = <V = unknown>({
     value,
   });
 };
+
+export const getStringListPatches = (
+  path: string,
+  currentList: string[],
+  newList: string[],
+  valueBuilder: (value: string[]) => unknown,
+) => {
+  const patches: PatchRequest = [];
+
+  const newLen = newList.length;
+  const curLen = currentList.length;
+
+  if (newLen === 0 && curLen > 0) {
+    patches.push({
+      path,
+      op: 'remove',
+    });
+  } else if (newLen > 0 && curLen === 0) {
+    patches.push({
+      path,
+      op: 'add',
+      value: valueBuilder(newList),
+    });
+  } else {
+    const hasDifferentItems = differenceWith(currentList, newList, isEqual).length > 0;
+
+    if (newLen !== curLen || hasDifferentItems) {
+      patches.push({
+        path,
+        op: 'replace',
+        value: valueBuilder(newList),
+      });
+    }
+  }
+
+  return patches;
+};
+
+export const getLabelPatches = (
+  basePath: string,
+  currentLabels: Record<string, string>,
+  newLabels: FlightCtlLabel[],
+) => {
+  const patches: PatchRequest = [];
+
+  const allKeys: string[] = Object.entries(currentLabels).map(([key]) => key);
+  newLabels.forEach((newLabel) => {
+    if (!allKeys.includes(newLabel.key)) {
+      allKeys.push(newLabel.key);
+    }
+  });
```

Comment:
I dont see this `allKeys` being used anywhere

Reply:
Yes, it was old code from a previous version.
Removed it.